x-pack/plugins/observability_solution/synthetics/server/types.ts

```diff
@@ -57,6 +57,7 @@ export interface SyntheticsServerSetup {
   uptimeEsClient: UptimeEsClient;
   basePath: IBasePath;
   isDev?: boolean;
+  isServerless?: boolean;
```

Comment:
Also here why do we define it as possible undefined? I think this should be just true/false

Reply:
Sure, this is fair. For flags like this that are based on config or some upstream dependency a lot of times I default to making them optional. But the code that sets this flag will always return a boolean, so it does make sense to make this field more opinionated. Just like the `isDev` flag above this one could also be optional.

I've also changed the assertions on this flag from general truthy/falsey to explicit boolean assertion.

8ddc360658faa4ff161b2f24829df4797ec950c2